### PR TITLE
Addition of io_k8s_repos and removal of vendor deps

### PR DESCRIPTION
**Pull request by chases2**
Allows use of @io_k8s_repo_infra//hack commands in bazel
Also runs automatic cleanup/update command
@io_k8s_repo_infra//hack:update-deps

Useful cleanup towards #2140

### CODE REVIEW

**Comment by michelle192837** [comment]
This looks good to me (once it passes tests), but will leave approval to people who contribute more often.

File: testgrid/istio.gen.yaml
Comment (fejta) [inline testgrid/istio.gen.yaml]: Is this intentional?

Comment (chases2) [reply testgrid/istio.gen.yaml]: No; this change should be in a different PR. This is what I was working on before

**Comment by chases2** [comment]
Update: ran `make gen`
Keeping as different commits to illustrate the difference between `make gen` and `bazel run @io_k8s_repo_infra//hack:update-deps`

**Comment by fejta** [comment]
Do we still need this?

**Comment by chases2** [comment]
I don't think we _need_ this, but the next person who comes by to try and use Bazel will certainly solve the same problem.

Giving Bazel a ref to the `io_k8s_repo_infra` makes using Bazel much easier, and I don't think it costs other workflows (ie. make) anything.

**Comment by chases2** [comment]
TL;DR I would still like to merge this

**Comment by fejta** [comment]
Will this pass unit tests? aka `bazel test //...` works?

**Comment by chases2** [comment]
No, although `bazel test //...` didn't pass before, either.

Both with and without changes, `bazel test //...` is failing with similar errors:
Before:
```
ERROR: /usr/local/google/home/slchase/repositories/istio-test-infra/prow/config/BUILD.bazel:29:1: no such package 'vendor/github.com/ghodss/yaml': BUILD file not found on package path and referenced by '//prow/config:go_default_library'
ERROR: Analysis of target '//prow/config:go_default_library' failed; build aborted: no such package 'vendor/github.com/ghodss/yaml': BUILD file not found on package path
```
After:
```
ERROR: /usr/local/google/home/slchase/repositories/istio-test-infra/prow/config/BUILD.bazel:8:1: no such package 'prow/cluster': BUILD file not found in any of the following directories.
 - /usr/local/google/home/slchase/repositories/istio-test-infra/prow/cluster and referenced by '//prow/config:go_default_test'
ERROR: Analysis of target '//prow/config:go_default_test' failed; build aborted: no such package 'prow/cluster': BUILD file not found in any of the following directories.
```

**Comment by chases2** [comment]
What this does let us do is run `@io_k8s_repo_infra//hack:update-deps`, `@io_k8s_repo_infra//hack:update-bazel`, etc.

**Review by fejta — APPROVED** [review]
/hold
/assign @clarketm @howardjohn 
fyi

**Pull request merged**
